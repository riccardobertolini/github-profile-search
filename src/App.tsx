import React, { useState } from 'react';
import { SearchBar } from './components/SearchBar';
import { UsersList, items } from './components/UsersList';
import './style/App.css';

interface defaultList {
    items: [
        items
    ],
}

export const App = () => {
  const [input, setInput] = useState('');
  const [githubUserList, setGithubUserList] = useState<defaultList>();
  
  const fetchData = async (input:string) => {
    return await fetch(`https://api.github.com/search/users?q=${input}&per_page=20&page=1`)
        .then(response => response.json())
        .then(data => {
            setGithubUserList(data)
        });
  }

  const updateSearch = async (input:string) => {
    fetchData(input)
    setInput(input);
  }
  
  
    return (
      <div className={'app'}>
        <img src={'/github_logo.png'} alt="GitHub Logo" />
        <h1>Users search</h1>
          <p>Navigate using keyboard arrow â†‘ and â†“, then press ENTER to open the profile link, or just click on it.</p>
        <SearchBar
            input={input}
            onChange={updateSearch}
        />
          {githubUserList && <UsersList usersList={githubUserList} />}
          <div className="credit">Made with ðŸ–¤ by <a href={"https://github.com/riccardobertolini/"}>riccardobertolini</a></div>
      </div>
  );
}

export default App;
