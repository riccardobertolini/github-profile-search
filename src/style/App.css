.app {
  max-width: 600px;
  width: 100%;
  margin: 30px auto;
  text-align: center;
  font-family: sans-serif;
}

.app img {
  width: 100%;
  max-width: 500px;
}

.app h1 {
  text-transform: uppercase;
  font-size: 20px;
  padding-bottom: 40px;
  border-bottom: 1px solid #000;
}

.app p {
  padding: 20px;
  border: 1px solid #999;
  margin: 40px;
}

.credit {
  margin-top: 50px;
  font-size: 12px;
}

.credit a,
.credit a:visited {
  color: black;
}