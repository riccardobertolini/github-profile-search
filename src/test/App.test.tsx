import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

test('app should render', () => {
  const wrapper = render(<App />);
  
  expect(wrapper).toMatchSnapshot();
});
