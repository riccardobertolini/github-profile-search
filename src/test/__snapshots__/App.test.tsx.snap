// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <img
          alt="GitHub Logo"
          src="/github_logo.png"
        />
        <h1>
          Users search
        </h1>
        <p>
          Navigate using keyboard arrow â†‘ and â†“, then press ENTER to open the profile link, or just click on it.
        </p>
        <input
          placeholder="Search a profile"
          style="width: 100%; background: white; border: 1px solid #999; padding: 0.5rem; height: 40px; font-size: 20px;"
          value=""
        />
        <div
          class="credit"
        >
          Made with ðŸ–¤ by 
          <a
            href="https://github.com/riccardobertolini/"
          >
            riccardobertolini
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <img
        alt="GitHub Logo"
        src="/github_logo.png"
      />
      <h1>
        Users search
      </h1>
      <p>
        Navigate using keyboard arrow â†‘ and â†“, then press ENTER to open the profile link, or just click on it.
      </p>
      <input
        placeholder="Search a profile"
        style="width: 100%; background: white; border: 1px solid #999; padding: 0.5rem; height: 40px; font-size: 20px;"
        value=""
      />
      <div
        class="credit"
      >
        Made with ðŸ–¤ by 
        <a
          href="https://github.com/riccardobertolini/"
        >
          riccardobertolini
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
